hack/lib/util/sync.sh

```
@@ -0,0 +1,110 @@
+    local fromDir="$4"
+    local toRepo="$5"
+    local toBranch="$6"
+    local fromRepoName=$(basename ${fromRepo})
+    local toRepoName=$(basename ${toRepo})
+
+    fromRepoLocation="${tmpDir}/${fromRepoName}"
+    toRepoLocation="${tmpDir}/${toRepoName}"
+
+    mkdir -p "${fromRepoLocation}"
+    mkdir -p "${toRepoLocation}"
+
+    pushd "${fromRepoLocation}" > /dev/null
+    # if we already have a git repo, assume it's the right one and get the requested branch
+    if ! git rev-parse --quiet --is-inside-work-tree &> /dev/null; then
+        git clone --quiet --single-branch --branch="${fromBranch}" "${fromRepo}" "${fromRepoLocation}"
+    fi
+    git fetch --refmap -- origin refs/heads/"${fromBranch}":refs/remotes/origin/"${fromBranch}"
+    git checkout --quiet "${fromBranch}"
+    git reset --quiet --hard origin/"${fromBranch}"
+    firstFromSHA=$(git rev-list --max-parents=0 HEAD)
+    newFromSHA=$(git log --oneline --format='%H' -1)
+    popd > /dev/null
+
+    pushd "${toRepoLocation}" > /dev/null
+    # if we aren't in a git repo, clone it
+    if ! git rev-parse --quiet --is-inside-work-tree &> /dev/null; then
+        git clone --quiet --single-branch --branch="${toBranch}" "${toRepo}" "${toRepoLocation}"
+    fi
+    git fetch --refmap -- origin refs/heads/"${toBranch}":refs/remotes/origin/"${toBranch}"
+    git checkout --quiet "${toBranch}"
+    git reset --quiet --hard origin/"${toBranch}"
+    startingFromSHA="${firstFromSHA}"
+    if [ -f "${fromRepoName}.sha" ]; then
+        startingFromSHA=$(cat "${fromRepoName}.sha")
+    fi
+    if [ "${newFromSHA}" == "${startingFromSHA}" ]; then
+        popd > /dev/null
+        echo "already at level: ${newFromSHA}, nothing to sync"
+        exit 0
+    fi
+
+    os::log::info ""
+    os::log::info "Syncing from ${startingFromSHA}..${newFromSHA}"
+
+    # create a clean branch to start from
+    set +o errexit
+    git branch --quiet -D "${toBranch}-working" &> /dev/null
+    set -o errexit
```

You can ignore error code of a single command by using `||`. For example:
```diff
-    set +o errexit
-    git branch --quiet -D "${toBranch}-working" &> /dev/null
-    set -o errexit
+    git branch --quiet -D "${toBranch}-working" &> /dev/null || true
```